### PR TITLE
OCPBUGS-16770: add need-management-kas-access label to olm-collect-profiles pods

### PR DESCRIPTION
**Pull request by sjenning**
https://issues.redhat.com/browse/OCPBUGS-16770

### CODE REVIEW

**Comment by sjenning** [comment]
/hold
until I can manually verify since the e2e doesn't catch this

**Comment by enxebre** [comment]
I assume there's not trivial way to e2e this here https://github.com/openshift/hypershift/blob/main/test/e2e/util/util.go#L776 given is a cron job

/lgtm
/cherry-pick release-4.13

File: control-plane-operator/controllers/hostedcontrolplane/olm/olm-collect-profiles.go [hunk 1]
```diff
@@ -21,6 +21,9 @@ var (
 func ReconcileCollectProfilesCronJob(cronJob *batchv1.CronJob, ownerRef config.OwnerRef, olmImage, namespace string) {
 	ownerRef.ApplyTo(cronJob)
 	cronJob.Spec = olmCollectProfilesCronJob.DeepCopy().Spec
+	cronJob.Spec.JobTemplate.ObjectMeta.Labels = map[string]string{
+		config.NeedManagementKASAccessLabel: "true",
```
Comment (enxebre) [inline control-plane-operator/controllers/hostedcontrolplane/olm/olm-collect-profiles.go:25]: Actually I'm not sure the job will be matched by the pod selector

policy.Spec.PodSelector = metav1.LabelSelector{
		MatchExpressions: []metav1.LabelSelectorRequirement{
			{
				Key:      NeedManagementKASAccessLabel,
				Operator: "DoesNotExist",
				Values:   nil,
			},
			{
				Key:      "name",
				Operator: "NotIn",
				Values:   []string{"aws-ebs-csi-driver-operator"},
			},
		},

**Pull request merged**